rmf_site_editor/src/site/sdf_exporter.rs

```diff
@@ -0,0 +1,366 @@
+                    let mut model_collisions = vec![];
+                    let mut model_visuals = vec![];
+                    // TODO(luca) don't do full descendant iter here or we might add twice?
+                    // Iterate through children and select all meshes
+                    for model_child in DescendantIter::new(&q_children, model) {
+                        if q_collisions.get(model_child).is_ok() {
+                            // Now iterate through the children of the collision and add them
+                            for entity in DescendantIter::new(&q_children, model_child) {
+                                let Some((mesh, _)) = get_mesh_and_material(entity) else {
+                                    continue;
+                                };
+                                let Ok(tf) = q_global_tfs.get(entity) else {
+                                    continue;
+                                };
+                                let mut tf = tf.compute_transform();
+                                tf.translation.z = tf.translation.z + **elevation;
+                                model_collisions.push(MeshData {
+                                    mesh,
+                                    material: None,
+                                    transform: Some(tf),
+                                });
+                            }
+                        } else if q_visuals.get(model_child).is_ok() {
+                            // Now iterate through the children of the visuals and add them
+                            for entity in DescendantIter::new(&q_children, model_child) {
+                                let Some((mesh, material)) = get_mesh_and_material(entity) else {
+                                    continue;
+                                };
+                                let Ok(tf) = q_global_tfs.get(entity) else {
+                                    continue;
+                                };
+                                let mut tf = tf.compute_transform();
+                                tf.translation.z = tf.translation.z + **elevation;
+                                model_visuals.push(MeshData {
+                                    mesh,
+                                    material: Some(material),
+                                    transform: Some(tf),
+                                });
+                            }
+                        }
+                    }
+                    if **is_static {
+                        // This is part of the static world, add it to the static mesh
+                        collision_data.extend(model_collisions);
+                        visual_data.extend(model_visuals);
+                    } else {
+                        // Create a new mesh for it
+                        let filename = format!("{}/{}_collision.glb", folder.display(), **name);
```

Comment:
This would work for non static models (the linked snippet) but wouldn't for all other places where we use this convention, specifically:

* All the static models in a level map to a single static mesh (we could possibly use the level's `SiteID` for it I guess, but not too clean).
* Each link of a double door has its own mesh (important for asymmetric doors), we only assign a single SiteID to doors so unclear what to name the two different meshes.
* Each link of each lift door has its own mesh, lift doors don't even have a `SiteID` so we would still need some sort of naming convention.

Is it OK to only use it for non static meshes or should we use a convention that works for all models?

Reply:
I forgot to mention that there is a TODO to mention this limitation [here](https://github.com/open-rmf/rmf_site/blob/665811c8a5799d8431985c9477dcff294872b96c/rmf_site_format/src/sdf.rs#L522-L531). Ideally I would have liked to use something derived from `AssetSource` rather than a `NameInSite` or even a `SiteID` to make sure that we do not duplicate multiple instances of the same mesh.

That being said I noticed an issue in the logic where I'm doing `<include>` in the world file rather than `<model>` tags and I'll be addressing it now.

Reply:
Issue addressed in https://github.com/open-rmf/rmf_site/pull/215/commits/65d9f8a265f2722fe584e92089d5d5d572183f9e, since I was doing `<include>` instead of actually spawning the meshes, it just happened to work because Gazebo itself already downloaded the model and added it to the path, not because the exported world was creating it

Reply:
> Each link of a double door has its own mesh (important for asymmetric doors), we only assign a single SiteID to doors so unclear what to name the two different meshes.

I think doing `"{site_id}_{segment_name}"` would be fine for this. That's the same as how it's currently done, just switching from `door_name` to `site_id`. There's less of an advantage here when the door type gets changed; the old segment types could still pollute the export directory, but I don't mind making it the user's responsibility to decide when to clear out their export directory.

Reply:
#219 merged after a bit of testing and a small fix, marking this as resolved